controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -124,3 +133,194 @@ func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocs
+	enableRbdMirrorDebugLoggingCommands :=
+		`echo "starting configuration of mirror logging"
+ceph config set client.rbd-mirror.a debug_ms 1
+ceph config set client.rbd-mirror.a debug_rbd 15
+ceph config set client.rbd-mirror.a debug_rbd_mirror 30
+ceph config set client.rbd-mirror.a log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set client.rbd-mirror-peer debug_ms 1
+ceph config set client.rbd-mirror-peer debug_rbd 15
+ceph config set client.rbd-mirror-peer debug_rbd_mirror 30
+ceph config set client.rbd-mirror-peer log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set mgr mgr/rbd_support/log_level debug
+		echo "completed"`
+	disableRbdMirrorDebugLoggingCommands :=
+		`echo "Removing configuration of mirror logging"
+ceph config rm client.rbd-mirror.a debug_ms
+ceph config rm client.rbd-mirror.a debug_rbd
+ceph config rm client.rbd-mirror.a debug_rbd_mirror
+ceph config rm client.rbd-mirror.a log_file
+ceph config rm client.rbd-mirror-peer debug_ms
+ceph config rm client.rbd-mirror-peer debug_rbd
+ceph config rm client.rbd-mirror-peer debug_rbd_mirror
+ceph config rm client.rbd-mirror-peer log_file
+ceph config rm mgr mgr/rbd_support/log_level
+		echo "completed"`
+
+	// Mirrooring is enabled but the debug logging is not enabled
+	if sc.Spec.Mirroring.Enabled && !rbdMirrorDebugLoggingEnabled {
+
+		err := r.Client.Create(r.ctx, getRbdMirrorDebugLoggingJob(sc, enableRbdMirrorDebugLoggingJobName, enableRbdMirrorDebugLoggingCommands))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return reconcile.Result{}, err
+		}
+		// Check if the job has succeeded
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, types.NamespacedName{Name: enableRbdMirrorDebugLoggingJobName, Namespace: sc.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if job.Status.Succeeded == 0 {
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		r.Log.Info("Successfully enabled rbd mirror debug logging")
+		rbdMirrorDebugLoggingEnabled = true
+	}
+	// Mirrooring is not enabled but the debug logging is enabled
+	if !sc.Spec.Mirroring.Enabled && rbdMirrorDebugLoggingEnabled {
+		err := r.Client.Create(r.ctx, getRbdMirrorDebugLoggingJob(sc, disableRbdMirrorDebugLoggingJobName, disableRbdMirrorDebugLoggingCommands))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return reconcile.Result{}, err
+		}
+		// Check if the job has succeeded
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, types.NamespacedName{Name: disableRbdMirrorDebugLoggingJobName, Namespace: sc.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if job.Status.Succeeded == 0 {
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+
+		// Delete both the enable & disable jobs
+		err = r.Client.Delete(r.ctx, getRbdMirrorDebugLoggingJob(sc, disableRbdMirrorDebugLoggingJobName, disableRbdMirrorDebugLoggingCommands))
```

Comment:
We don't need to call the whole func, We just need a kind with name and namespace set to delete an object.